packages/smooth_app/lib/data_models/fetched_product.dart

```diff
@@ -60,4 +61,29 @@ class FetchedProduct {
 
   /// When relevant, host of the query that we couldn't even ping.
   final String? failedPingedHost;
+
+  String getErrorTitle(
+    final AppLocalizations appLocalizations,
+  ) {
+    switch (status) {
+      case FetchedProductStatus.ok:
+        return 'Not supposed to happen...';
```

Comment:
Log some kind of error maybe?

Reply:
I agree it's lousy, but I couldn't think of something very elegant to deal with those cases: "there's no product, then show me the error". There's no error to display, hence the dummy `case`s.
I don't believe a sentry log would help here, but I'll happily add one if users complain about `'Not supposed to happen...'` titles.